File: build_multipart_form_data.php
function multipart_build_query($fields)
{
    $boundary = md5(time());
    $data = '';
    foreach ($fields as $field) {
        // add boundary
        $data .= '--' . $boundary . "\r\n";
        // add headers
        foreach ($field['headers'] as $header => $value) {
            $data .= $header . ': ' . $value . "\r\n";
        }
        // add blank line
        $data .= "\r\n";

        // add body
        $data .= $field['body'] . "\r\n";
    }
    // add closing boundary if there where fields
    if ($data) {
        $data .= $data .= '--' . $boundary . "--\r\n";
    }
    return $data;
}

File: build_multipart_form_data.py
#!/usr/bin/python
# Filename: build_multipart_form_data.py

import hashlib
import datetime

def build_multipart_form_data(fields):
    boundary = hashlib.md5(str(datetime.datetime.now())).hexdigest();
    data = '';
    for field in fields:
        #add boundary
        data += "--%s\r\n" % boundary
        # add headers
        for key, value in field['headers'].iteritems():
            data += "%s:%s\r\n" %(key, value)
        # add blank line
        data += "\r\n"
        # add body
        data += "%s\r\n" %field['body']
    # add closing boundary if there were fields
    if data:
        data += "%s--%s--\r\n" %(data, boundary)
    return boundary, data